R/fetch_description.R

```diff
@@ -168,3 +171,104 @@ fetch_description <-
 
     return(output)
   }
+
+
+
+#'Download (and potentially save) a description of a survey in Qualtrics Survey
+#'Format (QSF)
+#'
+#'@param surveyID String. Unique ID for the survey you want to download.
+#'  Returned as "id" by the [all_surveys] function.
+#'@param formatting String. Adjustments to the downloaded QSF JSON to be either
+#'  more human-readable or more compact.  Options are "none" (default, no
+#'  adjustment), "pretty" (more human-readable, using
+#'  \code{\link[jsonlite]{prettify}}), or "compact" (more compact, using
+#'  \code{\link[jsonlite]{minify}})
+#'@param save Boolean. Should downloaded QSF be output to R as a json object
+#'  (\code{FALSE}, default), or to a file (\code{TRUE})? Even when TRUE,
+#'  function still invisibly returns results.
+#'@param file String. File to write to (including path if needed).  Typically,
+#'  would end in .qsf, though not required.  Ignored if \code{save == FALSE}.
+#'
+#'
+#'@template retry-advice
+#'@return A JSON object equivalent of a Qualtrics QSF.
+#'
+#'@importFrom jsonlite toJSON
+#'@importFrom jsonlite prettify
+#'@importFrom jsonlite minify
+#'@importFrom rlang arg_match
+#'
+#'
+#'@export
+#'
+fetch_qsf <-
+  function(surveyID,
+           formatting = c("none", "pretty", "compact"),
+           save = FALSE,
+           file = NULL
+  ) {
```

Comment:
What if the signature was:

```r
write_qsf <- function (surveyID, file, formatting = c("none", "pretty", "compact") {}
```

Reply:
Agreed--in the initial design I wasn't focused on saving to disk, but with that focus this this argument order makes more sense.